File: backend/src/main/java/tn/itbs/backend/entites/Patient.java
package tn.itbs.backend.entites;


import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idPatient;

    @Column(nullable = false)
    private String nom;

    @Column(nullable = false)
    private LocalDate dateDeNaissance;

    @Column(nullable = false, unique = true)
    private String numTel;

    @OneToMany(mappedBy = "patient")
    private List<RendezVous> listP = new ArrayList<>();
}

File: frontend/frontend/src/app/features/patients/services/patient.ts
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class Patient {
  nom: string;
  dateDeNaissance: string;
  numTel: string;

  constructor() {
    this.nom = '';
    this.dateDeNaissance = '';
    this.numTel = '';
  }
}

